src/Control/Director.php

```diff
@@ -494,18 +515,11 @@ public static function host()
             }
         }
 
-        // Validate proxy-specific headers
-        if (TRUSTED_PROXY) {
-            // Check headers to validate
-            $headers = getenv('SS_TRUSTED_PROXY_HOST_HEADER')
-                ? explode(',', getenv('SS_TRUSTED_PROXY_HOST_HEADER'))
-                : ['HTTP_X_FORWARDED_HOST']; // Backwards compatible defaults
-            foreach ($headers as $header) {
-                if (!empty($_SERVER[$header])) {
-                    // Get the first host, in case there's multiple separated through commas
-                    return strtok($_SERVER[$header], ',');
-                }
-            }
+        if (!$request) {
+            $request = Injector::inst()->get(HTTPRequest::class, true, ['GET', '/']);
```

Comment:
One thing I've found is that sometimes we don't want get() to force a new object if none exists. I'm going to substitute with an Injector::has() check to ensure we only use the registered instance.